modules/widgets/social-icons.php

```diff
@@ -0,0 +1,500 @@
+	 *
+	 * See https://core.trac.wordpress.org/ticket/38387.
+	 */
+	$svg .= ' <use href="#icon-' . esc_html( $args['icon'] ) . '" xlink:href="#icon-' . esc_html( $args['icon'] ) . '"></use> ';
+
+	// Add some markup to use as a fallback for browsers that do not support SVGs.
+	if ( $args['fallback'] ) {
+		$svg .= '<span class="svg-fallback icon-' . esc_attr( $args['icon'] ) . '"></span>';
+	}
+
+	$svg .= '</svg>';
+
+	return $svg;
+}
+endif;
+
+if ( ! function_exists( 'jetpack_social_menu_social_links_icons' ) ) :
+/**
+ * Returns an array of supported social links (URL and icon name).
+ *
+ * @return array $social_links_icons
+ */
+function jetpack_social_menu_social_links_icons() {
+	// Supported social links icons.
+	$social_links_icons = array(
+		'500px.com'       => '500px',
+		'amazon.cn'       => 'amazon',
+		'amazon.in'       => 'amazon',
+		'amazon.fr'       => 'amazon',
+		'amazon.de'       => 'amazon',
+		'amazon.it'       => 'amazon',
+		'amazon.nl'       => 'amazon',
+		'amazon.es'       => 'amazon',
+		'amazon.co'       => 'amazon',
+		'amazon.ca'       => 'amazon',
+		'amazon.com'      => 'amazon',
+		'apple.com'       => 'apple',
+		'itunes.com'      => 'apple',
+		'bandcamp.com'    => 'bandcamp',
+		'behance.net'     => 'behance',
+		'codepen.io'      => 'codepen',
+		'deviantart.com'  => 'deviantart',
+		'digg.com'        => 'digg',
+		'dribbble.com'    => 'dribbble',
+		'dropbox.com'     => 'dropbox',
+		'etsy.com'        => 'etsy',
+		'facebook.com'    => 'facebook',
+		'/feed/'          => 'feed',
+		'flickr.com'      => 'flickr',
+		'foursquare.com'  => 'foursquare',
+		'goodreads.com'   => 'goodreads',
+		'google.com'      => 'google',
+		'plus.google.com' => 'google-plus',
+		'github.com'      => 'github',
+		'instagram.com'   => 'instagram',
+		'linkedin.com'    => 'linkedin',
+		'mailto:'         => 'mail',
+		'meetup.com'      => 'meetup',
+		'medium.com'      => 'medium',
+		'pinterest.com'   => 'pinterest',
+		'getpocket.com'   => 'pocket',
+		'reddit.com'      => 'reddit',
+		'skype.com'       => 'skype',
+		'skype:'          => 'skype',
+		'slideshare.net'  => 'slideshare',
+		'snapchat.com'    => 'snapchat',
+		'soundcloud.com'  => 'soundcloud',
+		'spotify.com'     => 'spotify',
+		'stumbleupon.com' => 'stumbleupon',
+		'tumblr.com'      => 'tumblr',
+		'twitch.tv'       => 'twitch',
+		'twitter.com'     => 'twitter',
+		'vimeo.com'       => 'vimeo',
+		'vk.com'          => 'vk',
+		'wordpress.org'   => 'wordpress',
+		'wordpress.com'   => 'wordpress',
+		'yelp.com'        => 'yelp',
+		'youtube.com'     => 'youtube',
+	);
+
+	return $social_links_icons;
+}
+endif;
+
+if ( ! function_exists( 'jetpack_social_menu_social_links_labels' ) ) :
+/**
+ * Returns an array of supported social links (URL and label).
+ *
+ * @return array $social_links_labels
+ */
+function jetpack_social_menu_social_links_labels() {
```

Comment:
Just to keep these in sync with the prior function, it may be wiser to have one function that returns all providers as associative arrays, and then just `wp_list_pluck` to get whatever args we actually care about at the time.

With two separate functions, I think it'd be difficult to maintain.